### PR TITLE
add open options for csv files

### PR DESCRIPTION
**Pull request by michaelmattig**

### CODE REVIEW

**Comment by michaelmattig** [comment]
[Test OgrSource with CSV file](https://trello.com/c/x1b46cuS/438-test-ogrsource-with-csv-file)

File: operators/src/source/ogr_source.rs [hunk 1]
```diff
@@ -361,6 +364,50 @@ where
         }
     }
 
+    fn open_gdal_dataset(dataset_info: &OgrSourceDataset) -> Result<Dataset> {
+        // TODO: reliably detect CSV files or allow defining them as such in params
+        if let Some(ext) = dataset_info.file_name.extension() {
+            if ext == "csv" || ext == "tsv" {
+                let columns = dataset_info
+                    .columns
```
Comment (ChristianBeilschmidt) [inline operators/src/source/ogr_source.rs:372]: Maybe we can split that into two functions to make it better readable and then do something like this:
```rust
match dataset_info.file_name.extension().and_then(OsStr::to_str) {
    Some("csv") | Some("tsv") => open_csv_dataset(dataset_info),
    _ => Ok(Dataset::open(&dataset_info.file_name)?),
}
```

File: operators/src/source/ogr_source.rs [hunk 2]
```diff
@@ -361,6 +364,50 @@ where
...
+                        None,
+                    )?);
+                }
+
+                return Ok(Dataset::open_ex(
+                    &dataset_info.file_name,
+                    None,
+                    None,
+                    Some(&[
+                        &("GEOM_POSSIBLE_NAMES=".to_owned() + &columns.x),
```
Comment (ChristianBeilschmidt) [inline operators/src/source/ogr_source.rs:400]: I think you should use `format!` instead.

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]
Looks really good, I have some issues regarding readability when I'm thinking at debugging in the future :smile:.

File: operators/src/source/ogr_source.rs (hunk 1 above)
Comment (michaelmattig) [reply operators/src/source/ogr_source.rs]: 992dbbe0c995dc1ba9aef4947614ba1dbd93eb96

File: operators/src/source/ogr_source.rs (hunk 2 above)
Comment (michaelmattig) [reply operators/src/source/ogr_source.rs]: 992dbbe0c995dc1ba9aef4947614ba1dbd93eb96

**Review by ChristianBeilschmidt — APPROVED** [review]

**Pull request merged**